service/src/test/java/uk/gov/hmcts/reform/fpl/service/ActionCmoServiceTest.java

```diff
@@ -69,8 +118,59 @@ void shouldAddOrderActionToCaseDataAndCaseManagementOrderWhenApproved() {
     void shouldAddOrderActionToCaseDataButNotCaseManagementOrderWhenNotApproved() {
         CaseDetails caseDetails = CaseDetails.builder().data(new HashMap<>()).build();
 
-        service.prepareCaseDetailsForSubmission(caseDetails, CaseManagementOrder.builder().build(), false);
+        actionCmoService.prepareCaseDetailsForSubmission(caseDetails, CaseManagementOrder.builder().build(),
+            false);
 
         assertThat(caseDetails.getData()).containsOnlyKeys("orderAction");
     }
+
+    @Test
+    void shouldReturnEmptyStringWhenOrderActionIsNotPresentOnCMO() {
+        List<Element<HearingBooking>> hearingBookings = createHearingBookings(MOCK_DATE);
+        CaseManagementOrder caseManagementOrder = CaseManagementOrder.builder().build();
+
+        String label = actionCmoService.createNextHearingDateLabel(caseManagementOrder, hearingBookings);
+
+        assertThat(label).isEqualTo("");
+    }
+
+    @Test
+    void shouldFormatNextHearingBookingLabelWhenCMOOrderActionContainsMatchingUUID() {
+        List<Element<HearingBooking>> hearingBookings = createHearingBookings(MOCK_DATE);
+
+        CaseManagementOrder caseManagementOrder = CaseManagementOrder.builder()
```

Comment:
Could we extract to private method, what do you think? more of a nice to have so up to you if you think we should leave

Reply:
What just the empty builder? Nah I think it's fine being in the test.

Reply:
I thought lines 141 - 143.. but if you happy where it is then fine with me :)

Reply:
Ah sorry, yes I agree i've made the change.

Reply:
i will forgive you i guess ;)